📖 List provider implementations of MachinePools

mboersma opened the pull request:
**What this PR does / why we need it**:

Adds a markdown table of known implementors of `MachinePool`. 

This was a task we talked about at the first MP working group meeting. I visited all listed provider repos and grepped for `MachinePool`. (I hear GCP has an implementation in progress as well.)

**Which issue(s) this PR fixes**:

Refs #9005

/area documentation
/area machinepool

mboersma commented:
/cc @richardcase

sbueringer commented:
Thx!
/approve
/hold

Feel free to merge once you have all the reviews you want

musaprg commented on docs/book/src/tasks/experimental-features/machine-pools.md:
```diff
@@ -26,3 +26,15 @@ Although MachinePools provide a similar feature to MachineDeployments, MachinePo
+| --- | --- | --- | --- |
+| Amazon | `AWSManagedMachinePool` / `AWSMachinePool` | Implemented, MachinePoolMachines supported | https://cluster-api-aws.sigs.k8s.io/topics/machinepools.html|
+| Azure | `AzureASOManagedMachinePool` / `AzureManagedMachinePool` / `AzureMachinePool` | Implemented, MachinePoolMachines supported | https://capz.sigs.k8s.io/self-managed/machinepools |
+| Oracle | `OCIManagedMachinePool` / `OCIMachinePool` | Implemented | https://oracle.github.io/cluster-api-provider-oci/managed/managedcluster.html |
```
OCI provider appears to have support of MachinePoolMachines as `OCIMachinePoolMachine`.

https://github.com/oracle/cluster-api-provider-oci/pull/304

musaprg reviewed (commented):
I'd prefer to use the name listed in [the provider list](https://cluster-api.sigs.k8s.io/reference/providers), but no strong preference.

richardcase reviewed (commented):
Looks good, thanks for doing this @mboersma.

It would be good to add CAPG into the list as there is an implementation in progress.

richardcase commented on docs/book/src/tasks/experimental-features/machine-pools.md:
```diff
@@ -26,3 +26,15 @@ Although MachinePools provide a similar feature to MachineDeployments, MachinePo
+
+| Provider | Implementations | Status | Documentation |
+| --- | --- | --- | --- |
+| Amazon | `AWSManagedMachinePool` / `AWSMachinePool` | Implemented, MachinePoolMachines supported | https://cluster-api-aws.sigs.k8s.io/topics/machinepools.html|
```
I think this would be good.

mboersma commented:
I updated the providers to match how they're described elsewhere in the docs, noted that OCI supports MPMs, and added a section for the GCP implementation in progress. Thanks everyone for the helpful feedback!

mboersma commented on docs/book/src/tasks/experimental-features/machine-pools.md:
```diff
@@ -26,3 +26,15 @@ Although MachinePools provide a similar feature to MachineDeployments, MachinePo
+| Provider | Implementations | Status | Documentation |
+| --- | --- | --- | --- |
+| AWS | `AWSManagedMachinePool` / `AWSMachinePool` | Implemented, MachinePoolMachines supported | https://cluster-api-aws.sigs.k8s.io/topics/machinepools.html|
+| Azure | `AzureASOManagedMachinePool` / `AzureManagedMachinePool` / `AzureMachinePool` | Implemented, MachinePoolMachines supported | https://capz.sigs.k8s.io/self-managed/machinepools |
```
I think that's much better, thanks for noticing this!
<img width="2208" height="696" alt="image" src="https://github.com/user-attachments/assets/95fa1fd6-a457-4828-9520-293fe079b215" />
